chat/public/js/src/chat.js

```diff
@@ -361,9 +361,8 @@ function ChatXBlock(runtime, element, init_data) {
      * pause: delays execution with a timeout
      */
     var pause = function(timeout) {
-        return new Promise(function(resolve, reject) {
-            setTimeout(resolve, timeout);
-        });
+        var promise = $.Deferred();
+        setTimeout(promise.resolve, timeout);
```

Comment:
@mtyaka The `return` is still there, it's just the line below this comment chain. So unfortunately that's not the problem here! It is something to do with the `var submitResponse` changes I made - if I replace it with the code on master all the tests pass.